tests/test_coordinateConverter.py

```diff
@@ -0,0 +1,556 @@
+# This file is part of cbp.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import itertools
+import math
+import unittest
+
+import numpy as np
+
+from lsst.sphgeom import Vector3d
+from lsst.afw.cameraGeom import FIELD_ANGLE, FOCAL_PLANE
+from lsst.geom import Box2D, Point2D, SpherePoint, arcseconds, degrees, radians
+import lsst.cbp.coordUtils as coordUtils
+from lsst.cbp.testUtils import SampleCoordinateConverter
+import lsst.cbp.coordinateConverter
+import lsst.utils.tests
+
+# set True to record and report numeric error in convertVectorFromPupilToBase
+# and in the iterative component of setFocalFieldAngle
+ReportRecordedErrors = True
+
+
+class CoordConverterTestCase(lsst.utils.tests.TestCase):
+    def setUp(self):
+        if ReportRecordedErrors:
+            lsst.cbp.coordUtils.startRecordingErrors()
+            lsst.cbp.coordinateConverter.startRecordingErrors()
+        self.scc = SampleCoordinateConverter()
+        self.cco = self.scc.coordinateConverter
+        # a list of all detector names plus None for the default detector
+        self.detectorNames = itertools.chain([None], self.cco.cameraGeom.getNameIter())
+        # set values for maximum error that are "plenty good enough"
+        self.maxPupilPosErr = 1e-4  # mm
+        self.maxDetectorPosErr = 1e-4  # pixels
+        self.maxFocalPlanePosErr = self.scc.pixelSizeMm * self.maxDetectorPosErr  # mm
+        self.maxFieldAngleErrRad = (0.001*arcseconds).asRadians()
+
+    def tearDown(self):
+        if ReportRecordedErrors:
+            utilsErrList = coordUtils.getRecordedErrors()
+            coordUtils.stopRecordingErrors()
+            if len(utilsErrList) > 0:
+                print("\nWarning: recorded {} numerical errors in computeAzAltFromBasePupil;"
```

Comment:
The question is whether this should be a real python warning so that it gets captured as such by pytest (or at least be a print to stderr). The print will get captured by pytest but is generally hidden. I tend to think we shouldn't use log in test code.

Reply:
These warnings are expected and are intended primarily in case somebody wants to try to reduce numerical error in the future. Thus I am not keen to make them Python warnings.